controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -218,6 +267,201 @@ func TestProfileReconcile(t *testing.T) {
+		} else {
+			actualBlockPoolParameters := r.cephBlockPool.Spec.Parameters
+			assert.Equal(t, v1.StorageProfilePhaseRejected, rejectedStorageProfile.Status.Phase)
+			assert.Nil(t, actualBlockPoolParameters, caseLabel)
+		}
+	}
+
+}
+
+func TestStorageProfileCephFsSubVolGroup(t *testing.T) {
+	var err error
+	var caseCounter int
+
+	var primaryTestCases = []struct {
+		label             string
+		expectedGroupName string
+		failureExpected   bool
+		createObjects     []runtime.Object
+		cephResources     []*v1alpha1.CephResourcesSpec
+		storageProfile    *v1.StorageProfile
+		cephFs            *rookCephv1.CephFilesystem
+	}{
+		{
+			label:             "valid profile",
+			expectedGroupName: "test-subvolgroup",
+			storageProfile:    fakeStorageProfile,
+			cephFs:            fakeCephFs,
+			failureExpected:   false,
+			cephResources: []*v1alpha1.CephResourcesSpec{
+				{
+					Name: "test-subvolgroup",
+					Kind: "CephFilesystemSubVolumeGroup",
+				},
+			},
+			createObjects: []runtime.Object{
+				&rookCephv1.CephFilesystemSubVolumeGroup{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-subvolgroup",
+						Namespace: namespaceName,
+					},
+					Status: &rookCephv1.CephFilesystemSubVolumeGroupStatus{},
+				},
+			},
+		},
+		{
+			label:             "rejected profile",
+			expectedGroupName: "test-subvolgroup",
+			storageProfile:    rejectedStorageProfile,
+			cephFs:            fakeCephFs,
+			failureExpected:   true,
+			cephResources: []*v1alpha1.CephResourcesSpec{
+				{
+					Name: "test-subvolgroup",
+					Kind: "CephFilesystemSubVolumeGroup",
+				},
+			},
+			createObjects: []runtime.Object{
+				&rookCephv1.CephFilesystemSubVolumeGroup{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-subvolgroup",
+						Namespace: namespaceName,
+					},
+					Status: &rookCephv1.CephFilesystemSubVolumeGroupStatus{},
+				},
+			},
+		},
+	}
+
+	for _, c := range primaryTestCases {
+		caseCounter++
+		caseLabel := fmt.Sprintf("Case %d: %s", caseCounter, c.label)
+		fmt.Println(caseLabel)
+
+		r := createFakeReconciler(t)
+		if strings.Contains(c.label, "rejected") {
+			r.storageCluster.Spec.DefaultStorageProfile = rejectedStorageProfile.Name
+		}
+
+		r.StorageClassRequest.Status.CephResources = c.cephResources
+		r.StorageClassRequest.Spec.Type = "sharedfilesystem"
+		r.StorageClassRequest.Spec.StorageProfile = c.storageProfile.Name
+
+		c.createObjects = append(c.createObjects, c.cephFs)
+		c.createObjects = append(c.createObjects, c.storageProfile)
+		c.createObjects = append(c.createObjects, fakeStorageConsumer)
+		fakeClient := fake.NewClientBuilder().WithScheme(r.Scheme).WithRuntimeObjects(c.createObjects...)
+
+		r.Client = fakeClient.Build()
+		r.StorageClassRequest.Status.CephResources = c.cephResources
+
+		_, err = r.reconcilePhases()
+		if c.failureExpected {
+			assert.Error(t, err, caseLabel)
+			continue
+		}
+		assert.NoError(t, err, caseLabel)
```

Comment:
This code is not reachable in case failure, Is this how it should?

Reply:
In case `failureExpected` label of the iterated case is set to true then it hits the error assertion, otherwise, etc. What exactly is unreachable?

Reply:
Resolving as per @rchikatw's manual confirmation